src/App.tsx

```diff
@@ -15,48 +16,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+function sortGoods(goods: string[], sortType: SortType, isReversed: boolean) {
+  const preparedGoods = [...goods];
+
+  if (sortType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortType) {
+        case SortType.Length:
+          return good1.length - good2.length;
+
+        case SortType.Name:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState<SortType>(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortType, isReversed);
+  const reset = () => {
+    setSortType(SortType.Default);
+    setIsReversed(false);
+  };
+
+  const isShowReset = sortType || isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info',
+            { 'is-light': sortType !== SortType.Name })}
+          onClick={() => setSortType(SortType.Name)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success',
+            { 'is-light': sortType !== SortType.Length })}
+          onClick={() => setSortType(SortType.Length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
```suggestion
          onClick={() => setIsReversed((prevValue) => !prevValue)}
```
https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state